src/Bicep.LangServer/Handlers/BicepBuildParamsCommandHandler.cs

```diff
@@ -46,14 +46,15 @@ public override async Task<string> Handle(string bicepParamsFilePath, Cancellati
 
         private async Task<string> GenerateCompiledParametersFileAndReturnOutputMessage(string bicepParamsFilePath, DocumentUri documentUri)
         {
-            string compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepParamsFilePath, OutputFormatOption.Json);
-            string compiledFile = Path.GetFileName(compiledFilePath);
+            var compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepParamsFilePath, OutputFormatOption.Json);
+            var compiledFile = Path.GetFileName(compiledFilePath);
+            var extension = Path.GetExtension(compiledFilePath).ToLowerInvariant();
 
-            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
-            // If not, we'll fail the generate params.
-            if (File.Exists(compiledFilePath) && !TemplateIsParametersFile(File.ReadAllText(compiledFilePath)))
+            // If the template exists and has a .json extension and contains the Bicep metadata, fail the build params.
+            // If not, continue to update the file.
+            if (extension == LanguageConstants.JsonFileExtension && File.Exists(compiledFilePath) && !TemplateIsParametersFile(File.ReadAllText(compiledFilePath)))
```

Comment:
We shouldn't do case-sensitive extension comparisons. Have you looked into using one of the shared methods for this? E.g.

https://github.com/Azure/bicep/blob/aff3e9967ca712a21e3fac022638522952d728fa/src/Bicep.Core/FileSystem/PathHelper.cs#L185